src/routes/+page.svelte

```diff
@@ -26,4 +29,22 @@
 
 <ThemeChange />
 
-<CardBoard {cards} />
+<CardBoard 
+  {cards}
+  on:cardSelected={(data) => {
+    console.log("Card selected:", data.detail)
+    selectedCard = data.detail.card;
+  }}
```

Comment:
I'm not seeing the benefit of all the event dispatching...
For example, it feels better to bind:selectedCard here.